pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/QueryRunner.java

```diff
@@ -149,13 +150,14 @@ public void shutDown() {
    * <p>This execution entry point should be asynchronously called by the request handler and caller should not wait
    * for results/exceptions.</p>
    */
-  public void processQuery(DistributedStagePlan distributedStagePlan, Map<String, String> requestMetadata) {
+  public void processQuery(DistributedStagePlan distributedStagePlan, Map<String, String> requestMetadataOriginal) {
+    Map<String, String> requestMetadata = consolidateMetadata(
```

Comment:
We can actually call it `opChainMetadata`. We can call this method after reading the `requestId` and `timeoutMs` for clarity because that is always at request level